lib/puppet/type/f5_persistencecookie.rb

```diff
@@ -0,0 +1,53 @@
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/parameter/f5_name.rb'))
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_address.rb'))
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_availability_requirement.rb'))
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_connection_limit.rb'))
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_connection_rate_limit.rb'))
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_description.rb'))
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_health_monitors.rb'))
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_ratio.rb'))
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_state.rb'))
+
+require File.expand_path(File.join(File.dirname(__FILE__),'..','..','puppet/property/f5_truthy.rb'))
+
+
+Puppet::Type.newtype(:f5_persistencecookie) do
+  @doc = 'Manage cookie persistence objects'
+
+  apply_to_device
+  ensurable
+
+  newparam(:name, :parent => Puppet::Parameter::F5Name, :namevar => true)
+
+  newproperty(:description, :parent => Puppet::Property::F5Description)
+
+  newproperty(:method) do
+    desc "Fail Safe Action. Valid values are 'reboot' or 'restart-all'."
+    newvalues(:'insert', :'passive', 'rewrite')
+  end
+
+  newproperty(:cookie_name) do
+    desc "cookie_name."
+  end
+
+  newproperty(:httponly, :parent => Puppet::Property::F5truthy) do
+    truthy_property("Valid values are 'enabled' or 'disabled'.")
```

Comment:
Having looked at `truthy_property` and its use in other types, proper booleans `true` and `false` should also work.  I think the string passed in should describe the property not just say `Valid values are 'enabled' or 'disabled'` each time.

Reply:
only 'enabled' or 'disabled' allowed

Reply:
No, the whole purpose of truthy_property is to allow true/false to be used.  truthy_property munges boolean-like values into enabled/disabled.

Reply:
@ericzji this is true.  @alexjfisher is correct in that you should let the f5_truthy property accept lazy values of what is effectively true or false.  This gives user flexibility which is helpful.